### PR TITLE
Teleport Release Table and cleaning up older docs

### LINKED ISSUE #3278: [Enhancement] Add Release Table to Teleport Docs
Teleport docs currently do not have a release table that discusses current releases, support dates, along with specified versions. 

Releases are currently listed here: https://github.com/gravitational/teleport/releases

Would be nice to have a `Releases` page in the docs, similiar to Gravity docs. Can look similar to this: https://gravitational.com/gravity/docs/changelog/#releases

### PR DESCRIPTION
**Pull request by benarent**
Fixes https://github.com/gravitational/teleport/issues/3278

This PR adds a release table to the FAQ and provides some guidelines for how the team updates and releases versions.  I've removed older versions and added them to a new repo as an archive. 
EOL Versions have been added to https://github.com/gravitational/documentation-archive 

This spring cleaning will help with upcoming documentation improvements.

### CODE REVIEW

**Comment by benarent** [comment]
This is the meat of the post. https://github.com/gravitational/teleport/blob/879993e82625e30c5bff654ec9b01bf381f369c2/docs/4.2/faq.md#which-version-of-teleport-is-supported


Release | LTS | Release Date | Supported Until | Minimum tsh version
-- | -- | -- | -- | --
4.2 | Yes | December 19th, 2019 | 4.3 is released | 3.x
4.1 | Yes | October 1st, 2019 | October 1st, 2020 | 3.x
4.0 | Yes | June 18th, 2019 | June 18th, 2020 | 3.x
3.2 | Yes | April 1st, 2019 | April 1st, 2020 | 2.x
3.1 | Yes | December 12th, 2018 | December 12th, 2019 | 2.x

Teleport uses a semantic versioning to release updates of software. With releases taking around 3-4 months, for MINOR releases. The team has these informal rules about releasing software.

 - Teleport Team will do it’s best to allow for a non-breaking upgrade even with MAJOR releases.
 - We aim to support versions for 3 releases (current and two back) or around 9 months. This means critical security fixes will be backported to earlier versions.
 - We provide Enterprise customers edge case support for older versions of Operating Systems. 
 - This might be a specific patch for an OSS in maintenance. This is provided on a somewhat ad-hoc basis.
 - We won’t support Teleport on an OS Distro that has stopped receiving maintenance updates and is EOL.

File: docs/4.2/faq.md [hunk 1]
```diff
@@ -49,6 +49,28 @@ Teleport nodes connected to proxy server (IoT) | 2,000* | 2x 2-4 vCPUs/8GB RAM |
...
+around 3-4 months, for MINOR releases. The team has these informal rules about releasing software.
+
+- Teleport Team will do it’s best to allow for a non-breaking upgrade even with MAJOR releases. 
+- We aim to support versions for 3 releases (current and two back) or around 9 months. 
+This means critical security fixes will be backported to earlier versions. 
+- We provide Enterprise customers edge case support for older versions of Operating Systems. This might be a specific patch for an OSS in maintenance. This is provided on a somewhat ad-hoc basis.
+- We won’t support Teleport on an OS Distro that has stopped receiving maintenance updates and is EOL. 
+
+**How should I upgrade my cluster?**
+Please follow our standard guidelines to [upgrading](amin-guide.md#upgrading-teleport).
```
Comment (webvictim) [inline docs/4.2/faq.md:72]: either `to upgrade` or `for upgrading`

File: docs/4.2/faq.md [hunk 2]
```diff
@@ -49,6 +49,28 @@ Teleport nodes connected to proxy server (IoT) | 2,000* | 2x 2-4 vCPUs/8GB RAM |
...
+| 4.2           | Yes | December 19th, 2019  | 4.3 is released      | 3.x                  |
+| 4.1           | Yes | October 1st, 2019    | October 1st, 2020    | 3.x                  |
+| 4.0           | Yes | June 18th, 2019      | June 18th, 2020      | 3.x                  |
+| 3.2           | Yes | April 1st, 2019      | April 1st, 2020      | 2.x                  |
+| 3.1           | Yes | December 12th, 2018  | December 12th, 2019  | 2.x                  |
+
+Teleport uses a semantic versioning to release updates of software. With releases taking
+around 3-4 months, for MINOR releases. The team has these informal rules about releasing software.
+
+- Teleport Team will do it’s best to allow for a non-breaking upgrade even with MAJOR releases.
```
Comment (webvictim) [inline docs/4.2/faq.md:65]: `its best`

File: docs/4.2/faq.md [hunk 3]
```diff
@@ -49,6 +49,28 @@ Teleport nodes connected to proxy server (IoT) | 2,000* | 2x 2-4 vCPUs/8GB RAM |
...
+
+| Release       | LTS | Release Date         | Supported Until      | Minimum tsh version  | 
+| --------------|-----| -------------------- | -------------------- | -------------------- |
+| 4.2           | Yes | December 19th, 2019  | 4.3 is released      | 3.x                  |
+| 4.1           | Yes | October 1st, 2019    | October 1st, 2020    | 3.x                  |
+| 4.0           | Yes | June 18th, 2019      | June 18th, 2020      | 3.x                  |
+| 3.2           | Yes | April 1st, 2019      | April 1st, 2020      | 2.x                  |
+| 3.1           | Yes | December 12th, 2018  | December 12th, 2019  | 2.x                  |
+
+Teleport uses a semantic versioning to release updates of software. With releases taking
```
Comment (webvictim) [inline docs/4.2/faq.md:62]: `uses semantic versioning`

Comment (webvictim) [reply docs/4.2/faq.md:62]: Also, this sentence is a weird fragment. Maybe this:

`Teleport uses semantic versioning to release updates of software, with minor releases taking around 3-4 months. The team has these informal rules about releasing software:`

**Review by webvictim — APPROVED** [review]
Yay, red diffs!

If we're marking every release as LTS, is there any point in making a distinction?

File: docs/4.2/faq.md (hunk 1 above)
Comment (webvictim) [reply docs/4.2/faq.md:72]: also the link should go to `admin-guide.md`

**Review by russjones — APPROVED** [review]

File: docs/4.2/faq.md (hunk 3 above)
Comment (benarent) [reply docs/4.2/faq.md:62]: As always thanks for the review Gus.

**Comment by webvictim** [comment]
Awesome :+1:

**Pull request merged**